### DISCUSSION TITLE
How to extract Teff and log(g) from phometric fit

### DISCUSSION THREAD

**Discussion by cm775** [cylammarco/WDPhotTools#15] [Q&A] [answered]
Hi, how do I extract Teff and log(g) from the photometric fitting code? I'd like to be able to save these as part of an array of photometric fits of different stars, rather than have these be printed out in the probability density map images. Any help would be appreciated.

**Comment by cylammarco** [comment, marked as answer]
Hi @cm775 ,

A fitter object contains the following information:
```
>>> ftr.__dict__.keys()
dict_keys(['column_key', 'column_key_formatted', 'column_key_unit', 'column_key_wavelength', 'column_names', 'column_units', 'column_wavelengths', 'column_type', 'dtype', 'model_da', 'model_db', 'interpolator', 'fitting_params', 'results', 'best_fit_params', 'best_fit_mag', 'sampler', 'samples', 'extinction_convolved', 'rv', 'pivot_wavelengths'])
```

If you used the minimize or the least-sqare methods, the outputs from the minimizers are stored in `ftr.results`, while the sampler and samples used in the MCMC are in `ftr.sampler` and `ftr.samples` respectively. Regardless of the choice of the 3 methods, the `ftr.best_fit_params` contains the best fit solutions for H and He atmosphere models (if you fit both). If only one atmosphere type is fitted, the one that is not fitted would be empty, e.g. using a random example with 10 ± 0.1 mag in the three Gaia filters, fixed log(g) = 7.81 and fixed distance = 21.0±0.1 pc:

```
>>> ftr.best_fit_params
{'H': {'Mbol': 1.643426521137146, 'logg': 7.81, 'G3': 8.104652131262878, 'distance': 21.0, 'dist_mod': 1.6110964736695965, 'G3_BP': 7.9162691540705215, 'G3_RP': 8.472247099094847, 'Teff': 90182.32449978741, 'Av_G3': 0.2875855630578689, 'Av_G3_BP': 0.3290203528808617, 'Av_G3_RP': 0.17171267263238885, 'mass': 0.6907265388002148, 'age': 267018.6413865415, 'chi2': 122.84646855889463}, 'He': {}}
```

so `ftr.best_fit_params['H']['Teff']` will give the effective temperature or `90182.32449978741` in the above example.

Hope this helps.

**Reply by cm775** [reply]
Hi, thanks for getting back to me so quickly. I just tried this on a fresh copy of the example fitting of PSO J1801+6254, and both the printed best_fit_param and the density map give a Teff value of 3945 K and a logg of 7.88 rather than the listed values in of 3542 and 7.58 in the example. Are there any reasons this could change from just being ran on a different system? I'm using a direct copy of the listed code as an example.

**Reply by cylammarco** [reply]
@cm775 You have found another typo/outdated material for me, thanks! The figures in the README.md were not updated to reflect the example code. The set {T=3542, logg=7.58} is the solution for photometric fitting with the 5 Pan-STARRRS filters and 3 Gaia filters, which was the example I previously used.

Since then, I included the JHK photometry, where the solution is precisely what you have found: {T=3945, logg=7.88)

![image](https://user-images.githubusercontent.com/2545839/182873361-af790c07-257e-4036-b916-ac9b467ccbe4.png)

I will try to fix the README.md during the next iteration of improvement, which can hopefully happen before the workshop in ~10 days.

**Reply by cylammarco** [reply]
I have also noticed this bug as I rerun that example code...

https://github.com/cylammarco/WDPhotTools/issues/17

**Reply by cm775** [reply]
Hi, thanks for letting me know. Also found that when running on scipy 1.8.1 everything runs ok but when running on scipy 1.9 I get the following error: ufunc 'isnan' not supported for the input types, and the inputs could not be safely coerced to any supported types according to the casting rule ''safe''.
![Screenshot 2022-08-04 at 16 13 44](https://user-images.githubusercontent.com/108353761/182883204-97a21fe7-3306-4a0d-ac1d-af88629c178d.png)

**Reply by cylammarco** [reply]
@cm775 can you please comment on [https://github.com/cylammarco/WDPhotTools/issues/18](https://github.com/cylammarco/WDPhotTools/issues/18) the version of numpy you are using? numpy 1.23 broke a few things for me and I fixed them in v0.0.3. But I'm pretty sure that's not the end of it...

**Comment by cylammarco** [comment]
Added also to the readme https://github.com/cylammarco/WDPhotTools#retrieving-the-fitted-solutions
and readthedocs https://wdphottools.readthedocs.io/en/latest/tutorials/fitting.html#retrieving-the-fitted-solution-s
